flepimop/gempyor_pkg/tests/distributions/test_distributions_common.py

```diff
@@ -0,0 +1,94 @@
+import numpy as np
+import pytest
+from gempyor.distributions import (
+    BetaDistribution,
+    BinomialDistribution,
+    FixedDistribution,
+    GammaDistribution,
+    LognormalDistribution,
+    NormalDistribution,
+    PoissonDistribution,
+    TruncatedNormalDistribution,
+    UniformDistribution,
+    WeibullDistribution,
+)
+
+
+@pytest.mark.parametrize(
+    "dist",
+    [
+        BetaDistribution(alpha=2.0, beta=5.0),
+        BinomialDistribution(n=10, p=0.5),
+        GammaDistribution(shape=2.0, scale=1.5),
+        LognormalDistribution(meanlog=0.0, sdlog=1.0),
+        NormalDistribution(mu=2.3, sigma=4.5),
+        PoissonDistribution(lam=3.0),
+        TruncatedNormalDistribution(mean=5.0, sd=2.0, a=0.0, b=10.0),
+        UniformDistribution(low=-0.5, high=1.5),
+        WeibullDistribution(shape=2.5, scale=5.0),
+    ],
+    ids=[
+        "Beta",
+        "Binomial",
+        "Gamma",
+        "Lognormal",
+        "Normal",
+        "Poisson",
+        "TruncatedNormal",
+        "Uniform",
+        "Weibull",
+    ],
+)
+def test_distribution_abc_rng_behavior(dist) -> None:
+    rng1 = np.random.default_rng(seed=42)
+    sample1 = dist.sample(rng=rng1)
+    rng2 = np.random.default_rng(seed=42)
+    sample2 = dist.sample(rng=rng2)
+    np.testing.assert_array_equal(sample1, sample2)
+
+    sample3_default = dist.sample(size=10)
+    sample4_default = dist.sample(size=10)
+    assert not np.array_equal(sample3_default, sample4_default)
```

Comment:
1. This should be two separate test fixtures, one testing the results are the same with RNGs created with the same seed and one with results being different when RNGs not provided.
2. Can we swap out `np.testing.assert_array_equal(sample1, sample2)` with `assert np.array_equal(sample1, sample2)` for consistency.
3. Part of the benefit of the template method we're using is that we don't have to test specific implementations to know that their behavior should comply with our expectations. Could we simplify by creating a dummy distribution in this test file rather than testing all of the implementations? If not that's also fine, but worth considering.

Reply:
related - whether we test with dummy or the set, we can define the annotation part once, then apply repeatedly.